\documentclass{article}
\usepackage[sc]{mathpazo}
\usepackage[utf8]{inputenc}
\usepackage{cite}
\usepackage{geometry}
\geometry{verbose,tmargin=2.5cm,bmargin=2.5cm,lmargin=2.5cm,rmargin=2.5cm}
\setcounter{secnumdepth}{2}
\setcounter{tocdepth}{2}
\usepackage{url}
\usepackage[unicode=true,pdfusetitle,
 bookmarks=true,bookmarksnumbered=true,bookmarksopen=true,bookmarksopenlevel=2,
 breaklinks=false,pdfborder={0 0 1},backref=false,colorlinks=false]
 {hyperref}
\hypersetup{
 pdfstartview={XYZ null null 1}}
\usepackage{breakurl}

\usepackage{setspace}
\usepackage{relsize}
\frenchspacing
\bibliographystyle{plos2009}

\begin{document}

\title{Phylogeographic analysis of porcine epidemic diarrhea virus}

\author{Eamon O'Dea}

\maketitle

\section*{Introduction}

Genomic analyses
~\cite{huang2013,chen_isolation_2013,marthaler_complete_2013,wang_new_2014,vlasova2014,marthaler2014}
have shown that several strains of PEDV are circulating in the United
States, raising the prospect of a phylogeographic assessment of
transmission routes, as in
Refs.~\cite{nelson2011,lemey2014,magee2014}. Molecular work thus far,
however, has not provided evidence regarding the large-scale spread of
PEDV beyond a possible origin in China.

There is some evidence that one strain leads to infection with milder
severity and could serve as a vaccine candidate~\cite{wang_new_2014},
but work to develop effective vaccines for the related TGEV
(transmissible gastroenteritis virus) has had limited success
~\cite[p. 513]{zimmerman_diseases_2012}.

We build on the work of Refs.~\cite{lemey2014,magee2014}.

\section*{Methods}

\subsection*{Tree fitting}

We obtained a set of nucleotide sequences by searching NCBI's
Nucleotide database with the query ``Porcine epidemic diarrhea
virus[Organism] AND (country=USA OR country=Mexico OR
country=Canada)''. We refined these results to whole genome sequences
that had both ``collection\_date'' and ``country'' features in their
GenBank records. Further, we checked that the sequences were from
field
strains. Refs.~\cite{marthaler_complete_2013,hoang2013,huang2013,chen_isolation_2013,stevenson2013,vlasova2014,oka2014,wang_new_2014}
were used in this regard, and for sequence with accession KJ584361 we
relied on a comment in the GenBank record to verify that it was a
field strain. Those references and other fields in the GenBank records
were also used to determine the U.S. state of collection for some
sequences.

These sequences were aligned using mafft~\cite{katoh2013}. The
alignment was run through Gblocks~\cite{castresana2000} to identify
any very poorly conserved, potentially misleading segments. Gblocks
removed only regions with many gaps such as the ends of the alignment
and indels. These regions contained no shared derived characters
beyond the indels in the spike gene, which have described
previously~\cite{wang_new_2014}. Information about these indels was
incorporated by enforcing monophyly of the set of sequences containing
them when searching for the best-fitting tree. The alignments were
also checked visually for evidence of alignment errors using
CLUSTAL~X~\cite{thompson1997}.

We used BEAST~\cite{drummond2012} to obtain from our alignment of
sequences a posterior sample of evolutionary trees with branch lengths
measured in calendar years. Because PEDV in North America is believed
to have spread rapidly in 2013~\cite{lowe2014}, we used an exponential
growth coalescent model~\cite{griffiths1994,drummond2002} as a prior
density for the times of ancestral nodes of the tree. To allow for
random variation in substitution rates among lineages, we used a
relaxed molecular clock model~\cite{drummond2006} in which the
substitution rates of branches were lognormaly distributed. To allow
for random variation in substitution rates among sites, we assume the
rates are sampled from an approximation to the
gamma-distribution~\cite{yang1994}, with four discrete rate categories
in our approximation. To allow for variation in substitution rates
among all unique pairs of nucleotide bases, we used the general
time-reversible model of nucleotide
substitutions~\cite{rodriguez1990}.

In setting up our BEAST analysis, we generally deemed the default
priors in the configuration generator BEAUTi~\cite{drummond2012}
acceptable with one notable exception. We chose the doubling time
parameterization of the exponential growth model rather than the
growth rate parameterization, which had the effect of preventing
models with negative growth rates from being proposed. We have placed
code to reproduce our input files in a web archive (TODO make sure
full build is working, make archive, add citation).

We discarded the first 10 percent of samples as burn-in, and we
performed the following checks for signs that the remaining samples
had failed to adequately sample the true posterior distribution. Using
Tracer~\cite{tracer}, we confirmed visually that all marginal
distributions seemed fairly smooth and that all traces appeared to
have constant means and variances. We increased the length of the
chains until all effective sample sizes exceeded 200, and ended up
with a chain of 100 million states in which the minimum effective
sample size was 408.

We obtained a one-tree summary of our posterior sample of trees using
the treeanotator progam that is distributed with BEAST. This tree was
one of our posterior trees having the maximum clade credibility
topology, which is the topology with the highest geometric mean of
posterior probabilities for all of its clades.

\section*{Results}


\begin{figure}[h]
\begin{adjustwidth}{-2.25in}{0in}
%  \includegraphics{fig1}
\caption{{\bf.} }\label{llsurface}
\end{adjustwidth}
\end{figure}

<<load-results,include=FALSE>>=

load("../work-sshd/mle.RData")

answer <- 7

@ 

We find \Sexpr{answer} important variables.

\section*{Discussion}


\bibliography{pedv}


\end{document}
