\documentclass{article}
\usepackage[sc]{mathpazo}
\usepackage[utf8]{inputenc}
\usepackage{cite}
\usepackage{geometry}
\geometry{verbose,tmargin=2.5cm,bmargin=2.5cm,lmargin=2.5cm,rmargin=2.5cm}
\setcounter{secnumdepth}{2}
\setcounter{tocdepth}{2}
\usepackage{url}
\usepackage[unicode=true,pdfusetitle,
 bookmarks=true,bookmarksnumbered=true,bookmarksopen=true,bookmarksopenlevel=2,
 breaklinks=false,pdfborder={0 0 1},backref=false,colorlinks=false]
 {hyperref}
\hypersetup{
 pdfstartview={XYZ null null 1}}
\usepackage{breakurl}
\usepackage{bm}
\usepackage{setspace}
\usepackage{relsize}
\frenchspacing
\bibliographystyle{plos2009}

\begin{document}

\title{Phylogeographic analysis of porcine epidemic diarrhea virus in
  North America}

\author{Eamon O'Dea}

\maketitle

\section*{Introduction}

Genomic analyses
~\cite{huang2013,chen_isolation_2013,marthaler_complete_2013,wang_new_2014,vlasova2014,marthaler2014}
have shown that several strains of PEDV are circulating in the United
States, raising the prospect of a phylogeographic assessment of
transmission routes, as in
Refs.~\cite{nelson2011,lemey2014,magee2014}. Molecular work thus far,
however, has not provided evidence regarding the large-scale spread of
PEDV beyond a possible origin in China.

There is some evidence that one strain leads to infection with milder
severity and could serve as a vaccine candidate~\cite{wang_new_2014},
but work to develop effective vaccines for the related TGEV
(transmissible gastroenteritis virus) has had limited success
~\cite[p. 513]{zimmerman_diseases_2012}.

We build on the work of Refs.~\cite{lemey2014,magee2014}.

\section*{Methods}

<<load-results,include=FALSE>>=

load("mle.RData")

@

\subsection*{Phylogenetic estimation}

We obtained a set of nucleotide sequences by searching NCBI's
Nucleotide database with the query ``Porcine epidemic diarrhea
virus[Organism] AND (country=USA OR country=Mexico OR
country=Canada)''. We refined these results to whole genome sequences
that had both ``collection\_date'' and ``country'' features in their
GenBank records. Further, we checked that the sequences were from
field
strains. Refs.~\cite{marthaler_complete_2013,hoang2013,huang2013,chen_isolation_2013,stevenson2013,vlasova2014,oka2014,wang_new_2014}
were used in this regard, and for sequence with accession KJ584361 we
relied on a comment in the GenBank record to verify that it was a
field strain. Those references and other fields in the GenBank records
were also used to determine the U.S. state of collection for some
sequences.

These sequences were aligned using mafft~\cite{katoh2013}. The
alignment was run through Gblocks~\cite{castresana2000} to identify
any very poorly conserved, potentially misleading segments. Gblocks
removed only regions with many gaps such as the ends of the alignment
and indels. These regions contained no shared derived characters
beyond the indels in the spike gene, which have described
previously~\cite{wang_new_2014}. Information about these indels was
incorporated by enforcing monophyly of the set of sequences containing
them when searching for the best-fitting tree. The alignments were
also checked visually for evidence of alignment errors using
CLUSTAL~X~\cite{thompson1997}.

We used BEAST~\cite{drummond2012} to obtain from our alignment of
sequences a posterior sample of rooted trees with branch lengths
measured in calendar years. Because PEDV in North America is believed
to have spread rapidly in 2013~\cite{lowe2014}, we used an exponential
growth coalescent model~\cite{griffiths1994,drummond2002} as a prior
density for the times of ancestral nodes of the tree. To allow for
random variation in substitution rates among lineages, we used a
relaxed molecular clock model~\cite{drummond2006} in which the
substitution rates of branches were lognormaly distributed. To allow
for random variation in substitution rates among sites, we assume the
rates are sampled from an approximation to the
gamma-distribution~\cite{yang1994}, with four discrete rate categories
in our approximation. To allow for variation in substitution rates
among all unique pairs of nucleotide bases, we used the general
time-reversible model of nucleotide
substitutions~\cite{rodriguez1990}.

TODO describe source of clock rate prior, cite maybe woo et al to
explain why unlinking clock for S and rest.

In setting up our BEAST analysis, we generally deemed the default
priors in the configuration generator BEAUTi~\cite{drummond2012}
acceptable with one notable exception. We chose the doubling time
parameterization of the exponential growth model rather than the
growth rate parameterization, which had the effect of preventing
models with negative growth rates from being proposed. We have placed
code to reproduce our input files in a web archive (TODO make sure
full build is working, make archive, add citation).

We discarded the first 10 percent of samples as burn-in, and we
performed the following checks for signs that the remaining samples
had failed to adequately sample the true posterior distribution. Using
Tracer~\cite{tracer}, we confirmed visually that all marginal
distributions seemed fairly smooth and that all traces appeared to
have constant means and variances. We increased the length of the
chains until all effective sample sizes exceeded 200, and ended up
with a chain of 100 million states in which the minimum effective
sample size was 408.

We obtained a one-tree summary of our posterior sample of trees using
the treeanotator progam that is distributed with BEAST. This tree was
one of our posterior trees having the maximum clade credibility
topology, which is the topology with the highest geometric mean of
posterior probabilities for all of its clades. We used
DensiTree~\cite{bouckaert2010} to plot the full sample of trees and
thus illustrate the degree to which the tree was resolved.

\subsection*{Regression modeling of geographic transition rates}

In the next few paragraphs, we describe our approach to fitting the
sequence location data. It is a maximum-likelihood variant of the
fully Bayesian approach introduced in Ref.~\cite{lemey2014}.

We assume that our tree estimated from the molecular data accurately
represents the genealogy of our sampled sequences in time. Then we
assume that the virus' location at points along each of the branches
in the genealogy is determined by a continuous-time Markov chain with
$K$ discrete states. The infinitesimal rates of change for transition
probabilities in this Markov chain are collected in a $K \times K$
matrix $\mathbf{\Lambda}$, where $\Lambda_{i,j}$ gives the rate of
change from state $i$ to state $j$. We assume these infinitesimals are
independent of time. The diagonal elements are determined by the
mathematical requirement that each row sums to zero but the
off-diagonal elements may vary freely. For a given setting of
$\Lambda$, the likelihood of the observed data may be efficiently
calculated with the same dynamic programming algorithm used to
calculate the likelihood of a column in a sequence alignment for a
given substitution model.

For many data sets---ours included---there is not enough information to
reliably estimate all estimate pairwise transition rates
$\Lambda_{i,j}$. Further we are less interested in these rates than in
how they may depend on other variables related to mechanisms of spread
or data collection. Thus we employ a linear model to scale the space
of all pairwise transition rates down to the space spanned by a set of
$P$ predictive variables that we derived based on hypotheses of such
mechanisms. Let $x_{i,j,w}$ be the value for predictor $w$ that
corresponds to $\Lambda_{i,j}$, and let $\beta_w$ be the regression
coefficient for predictor $w$. Then our regression model is
\begin{equation}
  \log \Lambda_{i,j} = \sum_{w=1}^P x_{i,j,w} \beta_w.
\end{equation}

We found the maximum likelihood regression coefficients
$\hat{\bm{\beta}}$ with the \texttt{optim.rphast} function in the
rPhast package~\cite{rphast}, which is essentially an implementation
of the BFGS method~\cite{nocedal1999}. Confidence intervals were
obtained from a parametric bootstrap as follows. The Markov process
was initialized by choosing one of the $K$ locations uniformly at
random and assigning it to the root node of the tree. Then the
locations of all descendant nodes were simulated recursively by
sampling from the transition probabilities for location along each
branch, as determined by the maximum likelihood model. The output was
a set of locations for the tips of the tree. We employed the
\texttt{simulate.msa} function of rPhast to carry out \Sexpr{bs$R} of
these simulations. For each simulation replicate, we obtained maximum
likelihood estimates of the regression coefficients. Confidence
intervals for each coefficient were calculated as the interval spanned
by the middle 95\% of these bootstrap estimates.

\subsection*{Choice of predictors and geographic states}

In our application, we matched the states of the geographic model to
the unique collection locations of sequences. This choice gave us a
total of $K =\Sexpr{nchar(modAsym$alphabet)}$ unique model states,
each of which corresponded to an area governed by either a U.S. state
or the nation of Mexico.

For a predictor of movement from one state to another, we used the
logarithm of the estimated annual flow of swine transported for
feeding or breeding purposes. These estimates came from year 2001
government records of interstate livestock movements~\cite{ers2003}.

\subsection*{Reproducibility}

We have scripted the main steps of our analysis to promote
reproducibility. The Docker framework~\cite{boettiger2015} has been
used to script and package a setup of the required software
environment as well. These scripts have been archived on the web (TODO
add zenodo citation).

\section*{Results}

As a prerequisite to fitting our geographic model, we obtained a
posterior samples of evolutionary trees from a set of PEDV sequences
collected from outbreaks. See the Methods for
details. Figure~\ref{densitree} displays these trees. Although the
tree is not fully resolved, several deep and shallow parts of it are
and permit further analysis. The trees are generally consistent with
previous work~\cite{vlasova2014} in placing the common ancestor of the
S INDEL and non-S INDEL strains before April 2013, the current upper
bound on the importation date of these strains to the country.

\begin{figure}[h]
%  \includegraphics[width=\textwidth]{fig1}
  \caption{{\bf The resolution of clades and clade ages for the PEDV
      sequences was variable.}  All trees in the posterior sample are
    plotted with a transparent color to illustrate the probability
    densities of all clades and node ages. At intermediate depths of
    the tree, several conflicting clades have moderate levels of
    support producing a blurred image of webs and multifurcations. The
    node positions on the $x$ axis represent their occurrence in
    calendar time. The time interval between the oldest and newest
    tree tip is about 9 months, which provides an indication of the
    time scale. The S INDEL sequences form the monophyletic group at the
    bottom that diverges from the rest of the sequences at the root of
    the tree.}\label{densitree}
\end{figure}

To gain insight into drivers of the geographic spread of PEDV, we
fitted a regression model of geographic transition rates. As described
in the Methods, the regression model parameters were optimized to fit
with the observed collection locations and the estimated genealogy of
a set of virus sequences. We found that the logarithm of transport
flows had a positive effect on transition rates
(Fig.~\ref{llsurface}). A likelihood ratio test supported rejection of
the hypothesis that flows had no effect $(\chi^2_1 =
\Sexpr{signif(Dasym, 3)}, p<\Sexpr{10^ceiling(log10(1 - pchisq(Dasym,
  df=1)))})$.

\begin{figure}[h]
\includegraphics{fig2}
\caption{{\bf Flows have a positive effect on geographic transition
    rates.} The red point indicates maximum likelihood estimate for
  regression coefficients in our linear predictor of geographic
  transition rates and the error bars represent 95\% parametric
  bootstrap percentile confidence intervals.}\label{llsurface}
\end{figure}

\section*{Discussion}

Using time- and location-tagged sequences, we have fitted a simple
linear model to rates of interstate migration of PEDV. We found that
interstate shipment flows were a statistically significant predictor
of these migration rates. Simulations were used to generate confidence
intervals for the regression coefficient and also served to verify the
accuracy of the method for these data.

The previous work introducing the model we
used~\cite{lemey2014,magee2014}, did not include such a
verification. Although the reliability of likelihood-based estimation
of transition rates is well-established for molecular data, it was not
clear that they could perform well with the comparatively smaller
amounts of geographic information in current field data. Thus the
finding that these methods are reasonably accurate with simulated data
sets of a similar size and shape to a real data set supports the
continued development and application of these methods.

benefits of speed

regression for molecular substitution rates useful?





\bibliography{pedv}

\end{document}
